### PR TITLE
Configure Tilt

### PR DESCRIPTION
**Pull request by rouja**
## Purpose

Tilt makes running your cluster k8s in local, it is a great tool to dev or debug your cluster before deploying.

## Proposal

- [x] Add Tilt config
- [x] Improve secrets

## Stack

In order to have Tilt working you must have this stack on your machine:
- kubectl
- helm
- helmfile
- plugin helm-diff
- plugin helm-secrets
- kind
- docker

## How to run

```sh
# Setup the cluster (only the first time): 
make start-kind

# Wait all good, they all must be in `Running` or `Completed`: 
kubectl get po -A

# Start Tilt:
make tilt-up
```
Tilt interface: 
http://localhost:10350/

When all is green:
https://desk.127.0.0.1.nip.io

## Demo

image

image

### CODE REVIEW

**Comment by AntoLC** [comment]
Nice.
I don't think it should replace totally the dev environment by it but we should definitively have it in the project.

Could we move `Tiltfile` and `start-kind.sh` to the `/bin` folder ?

File: src/helm/env.d/dev/secrets.enc.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
-djangoSecretKey: ENC[AES256_GCM,data:06KBEHV/gBgGoB4DXf9yTU5XK1xP9OXfyKEiSdSghV8XIMon3o1ajSWN+WNMRHkRZuU=,iv:ZeP1X4pQF9fVm7quzzVXSm2CSLrqAizwZD5QFmNOoSc=,tag:Dm/b+6CfznSC+CdKj1SCYA==,type:str]
+djangoSecretKey: ENC[AES256_GCM,data:9fOtt8oesY2CUahg972UGldDrqqF6Fa1Tn+bKxNpMbfXppQtPY2Jfu4EWKAaqH07X00=,iv:OC0ggDgCcja6h4IK73jVXZGDE1qp5OJfeNg182DKxQ4=,tag:ITMAWmPxW8lNBvm2Xefw/Q==,type:str]
+djangoSuperUserPass: ENC[AES256_GCM,data:mkLVMnc=,iv:qYBUdUwJk422RVm23/6CUKubFtBL+lofynSnkJglNQk=,tag:Md5FPXwCe9kl5BkICHszzg==,type:str]
```
Comment (lebaudantoine) [inline src/helm/env.d/dev/secrets.enc.yaml:2]: I have a quick question: Could you clarify why we're adding the `djangoSuperUserPass` secret to the development values in this commit? I'm trying to understand its connection.

Additionally, how do we ensure that secrets remain hidden during deployments? Might be great to add it in commits body for our future ourselves

Does creating the secrets file mean we need to update the development `secrets.enc` values? And how does this impact the staging environment?

File: Dockerfile [hunk 2]
```diff
@@ -25,7 +25,7 @@ COPY ./src/frontend/packages/eslint-config-people/package.json ./packages/eslint
 RUN yarn --frozen-lockfile
 
 ### ---- Front-end builder image ----
-FROM node:20 as frontend-builder
+FROM node:20 as frontend-builder-1
```
Comment (lebaudantoine) [inline Dockerfile:28]: Just a suggestion: Could we use more descriptive suffixes instead of `-1` and `-2`? It might help make the code clearer.

File: Makefile [hunk 3]
```diff
@@ -323,3 +323,12 @@ frontend-i18n-generate: \
...
 .PHONY: frontend-i18n-compile
+
+# -- K8S
+build-k8s-cluster: ## build the kubernetes cluster using kind
+	./bin/start-kind.sh
+.PHONY: build-k8s-cluster
+
+start-tilt: ## start tilt - k8s local development
+	tilt up -f ./bin/Tiltfile
+.PHONY: build-k8s-cluster
```
Comment (lebaudantoine) [inline Makefile:334]: *`start-tilt` ?

File: Makefile [hunk 4]
```diff
@@ -323,3 +323,12 @@ frontend-i18n-generate: \
 frontend-i18n-compile: ## Format the crowin json files used deploy to the apps
 	cd $(PATH_FRONT) && yarn i18n:deploy
 .PHONY: frontend-i18n-compile
+
+# -- K8S
+build-k8s-cluster: ## build the kubernetes cluster using kind
+	./bin/start-kind.sh
+.PHONY: build-k8s-cluster
```
Comment (lebaudantoine) [inline Makefile:330]: Just a small note: Is `build` the most accurate verb to use here? Does the shell script actually build something, or would a different term be more precise?

File: src/helm/helmfile.yaml [hunk 5]
```diff
@@ -26,6 +26,7 @@ releases:
     values:
       - auth:
           password: pass
+        architecture: standalone
```
Comment (lebaudantoine) [inline src/helm/helmfile.yaml:29]: Do you know why we add this line?

File: bin/start-kind.sh [hunk 6]
```diff
@@ -0,0 +1,102 @@
+#!/bin/sh
```
Comment (lebaudantoine) [inline bin/start-kind.sh:1]: I will @rouja review this file.

File: bin/Tiltfile [hunk 7]
```diff
@@ -0,0 +1,71 @@
+load('ext://uibutton', 'cmd_button', 'bool_input', 'location')
```
Comment (lebaudantoine) [inline bin/Tiltfile:1]: I will @rouja review this file.

File: src/frontend/apps/desk/.env.dev [hunk 8]
```diff
@@ -0,0 +1 @@
+NEXT_PUBLIC_API_URL=https://desk.127.0.0.1.nip.io/api/v1.0/
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/.env.dev:1]: I’m curious about how Next handles having both an `env.dev` and an `env.development` file. Is this addition necessary? We already define an environment variable when running the frontend in `env.d/dev/values.desk.yaml.gotmpl` at line 63

**Review by lebaudantoine — COMMENTED** [review]
@rouja might be helpful answering some of my comments, as he initiated this PR. Thx for your great work @AntoLC and @rouja

File: src/helm/helmfile.yaml (hunk 5 above)
Comment (AntoLC) [reply src/helm/helmfile.yaml:29]: https://github.com/bitnami/charts/blob/main/bitnami/redis/README.md#standalone

File: Dockerfile (hunk 2 above)
Comment (AntoLC) [reply Dockerfile]: [fixup! 🔨(CI) add Tilt](https://github.com/numerique-gouv/people/pull/178/commits/77bf4271f39fcc03fc1f2bacc0812126772c2e2d)

File: Makefile (hunk 3 above)
Comment (AntoLC) [reply Makefile]: [fixup! 🔨(CI) add Tilt](https://github.com/numerique-gouv/people/pull/178/commits/77bf4271f39fcc03fc1f2bacc0812126772c2e2d)

File: Makefile (hunk 4 above)
Comment (AntoLC) [reply Makefile]: [fixup! 🔨(CI) add Tilt](https://github.com/numerique-gouv/people/pull/178/commits/77bf4271f39fcc03fc1f2bacc0812126772c2e2d)

File: src/helm/helmfile.yaml (hunk 5 above)
Comment (AntoLC) [reply src/helm/helmfile.yaml:29]: https://github.com/bitnami/charts/blob/main/bitnami/redis/README.md#standalone

File: bin/Tiltfile (hunk 7 above)
Comment (AntoLC) [reply bin/Tiltfile:1]: He imports some libs.

File: bin/start-kind.sh (hunk 6 above)
Comment (AntoLC) [reply bin/start-kind.sh:1]: /bin/sh is an executable representing the system shell and usually implemented as a symbolic link pointing to the executable for whichever shell is the system shell. The system shell is basically the default shell that the script should use. In last couple of years, Debian (and Ubuntu) decided to switch the system shell from bash to dash — a similar shell but lighter and much faster.

File: src/frontend/apps/desk/.env.dev (hunk 8 above)
Comment (AntoLC) [reply src/frontend/apps/desk/.env.dev]: Yes we don't need it.

File: bin/Tiltfile (hunk 7 above)
Comment (rouja) [reply bin/Tiltfile:1]: It's for cmd_button "Run database migration" for instance

**Review by lebaudantoine — APPROVED** [review]
